packages/dd-trace/src/priority_sampler.js

```diff
@@ -21,6 +21,8 @@ const AUTO_KEEP = ext.priority.AUTO_KEEP
 const USER_KEEP = ext.priority.USER_KEEP
 const DEFAULT_KEY = 'service:,env:'
 
+let isViaRule
```

Comment:
Refactoring the logic of `sample` and `isSampled` would be cleaner than a global.